zilliqa/src/message.rs

```diff
@@ -248,7 +248,26 @@ impl QuorumCertificate {
     // Verifying an aggregated signature is a case of verifying the aggregated public key
     // against the aggregated signature
     pub fn verify(&self, _public_keys: Vec<NodePublicKey>) -> bool {
-        true
+        // Select which public keys have gone into creating the signature
+        let public_keys = _public_keys
+            .into_iter()
+            .zip(self.cosigned.iter())
+            .filter_map(
+                |(public_key, cosigned)| {
+                    if *cosigned {
+                        Some(public_key)
+                    } else {
+                        None
+                    }
+                },
+            )
+            .collect::<Vec<_>>();
```

Comment:
nit: We can avoid some allocations by passing the `impl Iterator` directly to `verify_aggregate` (maybe in a later PR)